src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -114,29 +133,62 @@ module.exports = function<T, P, I>(config : HostConfig<T, P, I>) : Reconciler {
       // progress.
       const current = workInProgress.alternate;
       const next = completeWork(current, workInProgress);
+
+      // The work is now done. We don't need this anymore. This flags
+      // to the system not to redo any work here.
+      workInProgress.pendingProps = null;
+
+      // TODO: Stop using the parent for this purpose. I think this will break
+      // down in edge cases because when nodes are reused during bailouts, we
+      // don't know which of two parents was used. Instead we should maintain
+      // a temporary manual stack.
+      // $FlowFixMe: This downcast is not safe. It is intentionally an error.
+      const parent = workInProgress.parent;
```

Comment:
(Does the suppression comment mean maybe you lose typechecking on `parent` for the rest of this function? Maybe `const parent : Fiber = workInProgress.parent : $FlowFixMe;` would be better?)